### ISSUE TITLE
Getting NaN value as validation loss in results.csv while training yolov8.yaml(for nano scale) using custom dataset 

### ISSUE THREAD

**Issue by chandra-ps612** [ultralytics/ultralytics#9223]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Train, Val

### Bug
**Training information**
![Screenshot from 2024-03-22 15-37-28](https://github.com/ultralytics/ultralytics/assets/89622996/abead5e1-b4e5-43b0-8295-e15e2aa4cc15)
**results.csv report**
![Screenshot from 2024-03-22 15-40-55](https://github.com/ultralytics/ultralytics/assets/89622996/53799b47-29e8-46ca-a9d4-bb856cf6459f)

### Environment

```
Ultralytics YOLOv8.1.30 🚀 Python-3.10.11 torch-2.2.1+cu121 CUDA:0 (NVIDIA GeForce GTX 1650, 3901MiB)
Setup complete ✅ (4 CPUs, 11.6 GB RAM, 421.2/438.5 GB disk)

OS              Linux-5.19.0-46-generic-x86_64-with-glibc2.36
Environment     Linux
Python          3.10.11
Install         pip
RAM             11.64 GB
CPU             Intel Core(TM) i5-7400 3.00GHz
CUDA            12.1

matplotlib      ✅ 3.8.3>=3.3.0
opencv-python   ✅ 4.9.0.80>=4.6.0
pillow          ✅ 10.2.0>=7.1.2
pyyaml          ✅ 6.0.1>=5.3.1
requests        ✅ 2.31.0>=2.23.0
scipy           ✅ 1.12.0>=1.4.1
torch           ✅ 2.2.1>=1.8.0
torchvision     ✅ 0.17.1>=0.9.0
tqdm            ✅ 4.66.2>=4.64.0
psutil          ✅ 5.9.8
py-cpuinfo      ✅ 9.0.0
thop            ✅ 0.1.1-2209072238>=0.1.1
pandas          ✅ 2.2.1>=1.1.4
seaborn         ✅ 0.13.2>=0.11.0
```

### Minimal Reproducible Example
**Python**
```
import warnings
from PIL import Image
warnings.simplefilter('ignore', Image.DecompressionBombWarning)
from ultralytics import YOLO
model = YOLO('yolov8n.yaml')
results = model.train(batch=24, imgsz=416, amp=False, pretrained=False, optimizer=SGD, cache=False, workers=4, device=0, conf=0.6, lrf=0.1, mixup=0.1, copy_paste=0.1)
```
**Note:**
The first three lines are to disable  ```DecompressionBombWarning``` in context of ```pillow```

### Additional

I also tried PyTorch 2.2.1 (CUDA 11.8) for the above machine, but found the same issue.
Albeit, There is no such issue while working on Ubuntu 18.04 machine powered by Nvidia GeForce GTX 1050 Ti GPU

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by Burhan-Q** [comment]
@chandra-ps612 this is a common issue with `NVIDIA GeForce GTX 1650` or any `16xx`-series GPU. You can try disabling Automatic Mixed Precision (AMP) with `amp=False` when training, as some users have been able to resolve the `nan` issue this way. The source of this issue is between the hardware of the `16xx`-series GPUs and PyTorch, not anything Ultralytics related.

**Comment by chandra-ps612** [comment]
@Burhan-Q Thanks for the response
While training, i set ```amp=False``` as you can see it in  **Minimal Reproducible Example**, But it is not helping me out to fix it.

**Comment by chandra-ps612** [comment]
I have fix this issue for ```Ubuntu 22.04``` machine powered  by ```Nvidia GeForce GTX 1650``` GPU. 
**PyTorch**
```
pip install torch==1.12.1+cu102 torchvision==0.13.1+cu102 torchaudio==0.12.1 --extra-index-url https://download.pytorch.org/whl/cu102

torch == 1.12.1 + Cu 10.2
torchvision == 0.13.1 + Cu 10.2
torchaudio == 0.12.1 + Cu 10.2
torchmetrics == 0.8.0
```
**ultralytics package**
```
ultralytics == 8.1.22
```
```python == 3.10.11```

**Comment by glenn-jocher** [comment]
@chandra-ps612 great to hear you've resolved the issue! Downgrading to an older CUDA version with the specific PyTorch build seems to be an effective workaround for the GTX 1650 GPU. Thanks for sharing this solution with the community! 🌟 If anyone else encounters similar issues, trying out this configuration might be a good start. Feel free to reach out if you have more insights or face any further challenges. Happy coding! 😊
